subprojects/core-api/src/main/java/org/gradle/api/artifacts/transform/TransformAction.java

```diff
@@ -16,20 +16,42 @@
 
 package org.gradle.api.artifacts.transform;
 
+import org.gradle.api.Action;
 import org.gradle.api.Incubating;
 
 import javax.inject.Inject;
 
 /**
  * Interface for artifact transform actions.
  *
- * <p>Implementations must provide a public constructor.</p>
+ * <p>
+ *     A transform action implementation is an abstract class implementing the {@link #transform(TransformOutputs)} method.
+ *     A minimal implementation may look like this:
+ * </p>
  *
- * <p>Implementations can receive parameters by using annotated abstract getter methods.</p>
+ * <pre class='autoTested'>
+ * import org.gradle.api.artifacts.transform.TransformParameters
  *
- * <p>A property annotated with {@link InputArtifact} will receive the <em>input artifact</em> location, which is the file or directory that the transform should be applied to.
+ * public abstract class MyTransform implements TransformAction&lt;TransformParameters.None&gt; {
+ *     {@literal @}InputArtifact
+ *     abstract Provider&lt;FileSystemLocation&gt; getInputArtifact();
  *
- * <p>A property annotated with {@link InputArtifactDependencies} will receive the <em>dependencies</em> of its input artifact.
+ *     {@literal @}Override
+ *     public void transform(TransformOutputs outputs) {
+ *         File input = getInputArtifact().get().getAsFile();
+ *         File output = outputs.file(input.getName() + ".transformed");
+ *         // Do something to generate output from input
+ *     }
+ * }
+ * </pre>
+ *
+ * <ul>
+ *     <li>Do not implement {@link #getParameters()} in your class, the method will be implemented by Gradle.</li>
+ *     <li>Implementations may only have a default constructor.</li>
+ *     <li>Implementations can receive parameters by using annotated abstract getter methods.</li>
```

Comment:
This seems ambiguous to me. Are you saying that implementers must add annotated abstract getter methods in order to support parameters? If so, what should they be annotated with?

Reply:
There are two different kinds of parameters: `TransformParameters` and other parameters to the action like `@InputArtifact` and `@InputArtifactDependencies`. This comment is about the second kind.

I re-organized the Javadoc in https://github.com/gradle/gradle/pull/9525.